docs/src/main/sphinx/functions/setdigest.rst

```diff
@@ -0,0 +1,106 @@
+=========================
+Set Digest functions
+=========================
+
+Trino offers several functions that deal with
+`MinHash <https://en.wikipedia.org/wiki/MinHash>`_ technique.
+
+MinHash is a technique  which is employed to quickly estimate the  
+`Jaccard similarity coefficient <https://en.wikipedia.org/wiki/Jaccard_index>`_ 
+between two sets.
+
+One of the common applications of the MinHash technique is in data mining where it is employed for 
+detecting at scale near-duplicate web pages. By using this information, the search engines 
+efficiently avoid showing two pages which are nearly identical within the search results.
+
+Data structures
+---------------
+
+Trino implements Set Digest data sketches by encapsulating the following components:
+
+- `HyperLogLog <https://en.wikipedia.org/wiki/HyperLogLog>`_
+- `MinHash with a single hash function <http://en.wikipedia.org/wiki/MinHash#Variant_with_a_single_hash_function>`_
+
+The Hyperloglog structure is used for the aproximation of the distinct elements
+in the original set.
+
+The MinHash structure is used to store a low memory footprint signature of the original set. 
+The similarity of any two sets is estimated by comparing their signatures.
+
+The Trino type for this data structure is called ``setdigest``.
+
+Trino offers the ability to merge multiple Set Digest data sketches.
+
+
+Serialization
+-------------
+
+Data sketches can be serialized to and deserialized from ``varbinary``. This
+allows them to be stored for later use.  
+
+
+Functions
+---------
+
+.. function:: make_set_digest(x) -> setdigest
+
+    Composes all input values of ``x`` into a ``setdigest``.
+
+    ``x`` must be of long type.
```

Comment:
I'd like to give a less abstract example here to help the users who may think  that these functions are appealing for them to get started.

Let's say that we start with the sentence:

`The quick brown fox jumps over the lazy dog.`

Is there a method available in Trino to easily obtain the 5(k) word long shingles corresponding to the sentence:

```
The quick brown fox jumps
quick brown fox jumps over
brown fox jumps over the
fox jumps over the lazy
jumps over the lazy dog
```
?

On the shingles there could be applied `murmur3` hash and then they could get aggregated to a Set digest.